core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -498,8 +501,7 @@ private[ui] class JobPagedTable(
     sortColumn: String,
     desc: Boolean
   ) extends PagedTable[JobTableRowData] {
-  val parameterPath = UIUtils.prependBaseUri(basePath) + s"/$subPath/?" +
-    parameterOtherTable.mkString("&")
+  val parameterPath = basePath + s"/$subPath/?" + parameterOtherTable.mkString("&")
```

Comment:
Peripherally, `UIUtils.prependBaseUri()` is called on `basePath` before it's passed here, so this is redundant and breaks all the sorting links that rely on `parameterPath` when using yarn. I fixed it here since it was breaking my fix on yarn.